### PR TITLE
Refactor is-website-vulnerable.js

### PR DESCRIPTION
**Pull request by VictorZubr**
## Description
Hi, Liran. This is my first participation in opensource. I Hope my suggestion on refactoring a function will be helpful.

After successful lint check in my local repository, when I tried to to push my changes to remote repository, the npm error was appear:
```
20 error code ELIFECYCLE
21 error errno 1
22 error is-website-vulnerable@0.0.0-development lint: `eslint . && npm run lint:lockfile`
22 error Exit status 1
```
This error was related to arrow functions. I had to change arrow functions to functional expressions in all scripts so that this error disappears.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Refactor (non-breaking change which not adds functionality)

## Related Issue
[Feat: refactor bin/is-website-vulnerable.js](https://github.com/lirantal/is-website-vulnerable/issues/60)

## How Has This Been Tested?
I did not add new tests. All old tests were successful.

## Checklist:

- [ ] I have updated the documentation (if required).
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [ ] I added a picture of a cute animal cause it's fun

### CODE REVIEW

**Comment by lirantal** [comment]
Hi @VictorZubr 👋

Happy to have you here helping us, thanks!
I'll go through the code and provide my comments.

File: bin/get-url.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+/* eslint no-console: ["error", { allow: ["error"] }] */
+const debug = require('debug')('get-url')
+const { prompt } = require('enquirer')
+const isUrl = require('is-url-superb')
+const { Utils } = require('../index')
+
+const validate = function(url) {
```
Comment (lirantal) [inline bin/get-url.js]: how about we change this to function declaration?
e.g.

```
function validate(url) 
...
```

File: bin/get-url.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  if (!isUrl(url)) {
+    throw new Error('Given argument is an invalid URL')
+  }
+
+  return Utils.parseUrl(url)
+}
+
+const getUrl = async function(url, isJson) {
```
Comment (lirantal) [inline bin/get-url.js]: same as above comment, changing this to a function declaration

File: bin/is-website-vulnerable.js [hunk 3]
```diff
@@ -1,48 +1,35 @@
...
 const { Audit, RenderConsole, RenderJson, Utils } = require('../index')
+const getUrl = require('./get-url')
 
-let url = process.argv[2]
-debug(`received url argument: ${url}`)
-
-const isWindows = os.type() === 'Windows_NT'
-const isJson = !!argv.json
-const showProgressBar = !isJson && !isWindows
+const getStatus = () => {
```
Comment (lirantal) [inline bin/is-website-vulnerable.js]: Good job here!
How about we rename it to something more descriptive, maybe like `getEnvironment` or `detectEnvironment`? I think the latter makes more sense but leaving it up to you if those or other naming is better

File: bin/is-website-vulnerable.js [hunk 4]
```diff
@@ -1,48 +1,35 @@
...
-            message: 'Please provide a URL to scan:',
-            validate: input => input && input.length > 0 && isUrl(input)
-          })
-          url = response.url
-        } catch (_) {
-          throw new Error('URL input prompt failed')
-        }
-      }
-    }
+const getOptions = () => {
```
Comment (lirantal) [inline bin/is-website-vulnerable.js]: Same as above, we can probably name it in a more descriptive way, how about `getLighthouseOptions` ?

File: bin/is-website-vulnerable.js [hunk 5]
```diff
@@ -1,48 +1,35 @@
...
-      throw new Error('Given argument is an invalid URL')
-    }
+  return { lighthouseOpts, chromeOpts }
+}
 
-    url = Utils.parseUrl(url)
+;(async () => {
+  try {
+    const { isWindows, isJson, showProgressBar } = getStatus()
+    const url = await getUrl(process.argv[2], isJson)
```
Comment (lirantal) [inline bin/is-website-vulnerable.js]: how about we change that file and function name for getUrl to something more descriptive like `promptUrlInput` ?

File: index.js [hunk 6]
```diff
@@ -1,5 +1,3 @@
-'use strict'
```
Comment (lirantal) [inline index.js]: why was this removed?

Comment (lirantal) [inline index.js]: looks like this was removed from other files as well. can you explain why? otherwise please add back

File: src/RenderConsole.js [hunk 7]
```diff
@@ -38,10 +38,10 @@ module.exports = class RenderConsole {
         output += `
   Vulnerabilities:`
       }
-
-      vulnerabilitiesResults.details.items.forEach(vulnItem => {
+      const that = this
```
Comment (lirantal) [inline src/RenderConsole.js]: I don't like this style :)
can you please revert the change. Why did you need to do it to begin with?

File: bin/get-url.js [hunk 8]
```diff
@@ -0,0 +1,43 @@
+/* eslint no-console: ["error", { allow: ["error"] }] */
+const debug = require('debug')('get-url')
```
[the remark this answers is not in the document]
Comment (VictorZubr) [reply bin/get-url.js]: Ok. I understand

File: bin/get-url.js (hunk 1 above)
Comment (VictorZubr) [reply bin/get-url.js]: Good!

File: bin/get-url.js (hunk 2 above)
Comment (VictorZubr) [reply bin/get-url.js]: Ок

File: bin/is-website-vulnerable.js (hunk 3 above)
Comment (VictorZubr) [reply bin/is-website-vulnerable.js]: Ok. I agree with you

File: bin/is-website-vulnerable.js (hunk 4 above)
Comment (VictorZubr) [reply bin/is-website-vulnerable.js]: Ok.

File: bin/is-website-vulnerable.js (hunk 5 above)
Comment (VictorZubr) [reply bin/is-website-vulnerable.js]: Ok.

File: index.js (hunk 6 above)
Comment (VictorZubr) [reply index.js]: I can fix all of the above. But starting from this point I need to solve two problems.

First one.
When I clone the project, the linter already shows some errors.


`npm install
npm run lint

C:\var\www\html\is-website-vulnerable-master>npm run lint

> is-website-vulnerable@0.0.0-development lint C:\var\www\html\is-website-vulnerable-master
> eslint . && npm run lint:lockfile


C:\var\www\html\is-website-vulnerable-master\bin\is-website-vulnerable.js
   3:1   error  'use strict' is unnecessary when implied strict mode is enabled  strict
  25:9   error  Unexpected console statement                                     no-console
  32:23  error  Expected parentheses around arrow function argument              arrow-parens
  48:7   error  Unexpected console statement                                     no-console
  88:5   error  Unexpected console statement                                     no-console
  89:5   error  Unexpected console statement                                     no-console

C:\var\www\html\is-website-vulnerable-master\index.js
  1:1  error  'use strict' is unnecessary when implied strict mode is enabled  strict

C:\var\www\html\is-website-vulnerable-master\src\Audit.js
  2:1  error  'use strict' is unnecessary when implied strict mode is enabled  strict

C:\var\www\html\is-website-vulnerable-master\src\RenderConsole.js
   3:1   error  'use strict' is unnecessary when implied strict mode is enabled  strict
  16:5   error  Unexpected console statement                                     no-console
  42:52  error  Expected parentheses around arrow function argument              arrow-parens
  76:47  error  Expected parentheses around arrow function argument              arrow-parens

C:\var\www\html\is-website-vulnerable-master\src\RenderJson.js
   3:1   error  'use strict' is unnecessary when implied strict mode is enabled  strict
  15:5   error  Unexpected console statement                                     no-console
  27:52  error  Expected parentheses around arrow function argument              arrow-parens
  53:5   error  Unexpected var, use let or const instead                         no-var
  72:50  error  Expected parentheses around arrow function argument              arrow-parens

C:\var\www\html\is-website-vulnerable-master\src\Utils.js
   4:3   error  Expected method shorthand                            object-shorthand
  14:3   error  Expected method shorthand                            object-shorthand
  21:25  error  Expected parentheses around arrow function argument  arrow-parens
  32:3   error  Expected method shorthand                            object-shorthand
  43:3   error  Expected method shorthand                            object-shorthand

✖ 22 problems (22 errors, 0 warnings)
  16 errors and 0 warnings potentially fixable with the `--fix` option.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! is-website-vulnerable@0.0.0-development lint: `eslint . && npm run lint:lockfile`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the is-website-vulnerable@0.0.0-development lint script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\User\AppData\Roaming\npm-cache\_logs\2020-01-05T09_46_19_863Z-debug.log
`

Comment (lirantal) [reply index.js]: I believe you may have some global lint and prettier rules, either in your IDE or for the user that override the project in some way. I did a clean git clone && npm install && npm run lint and it passes just alright.

Comment (VictorZubr) [reply index.js]: I did the same without my IDE and everything works correctly for me! Thanks.
Now I need time to fix everything and I will do a pullrequest.

Comment (lirantal) [reply index.js]: Great! :-)

**Comment by VictorZubr** [comment]
I will sen new pullrequest
